### ISSUE TITLE
Create API endpoint team-members/index.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#145]
Create the API endpoint team-members/index.ts that will API route for getting all team members. Be sure to add the proper extension when creating the file. Here are the instructions:

## Issue Description

This issue is to create a new Next.js API endpoint for retrieving all team members. The endpoint will be a GET request and will be used to fetch all the team members from the database using Prisma. The endpoint will be written in TypeScript using ES6 syntax, such as arrow functions. The data returned from the database query must be validated using Zod.

### Task Details

1. Create a new file `src/pages/api/team-members/index.ts` for the API endpoint.

2. Import the required modules and types:

   - Import `NextApiRequest` and `NextApiResponse` from `next`.
   - Import `PrismaClient` from `@prisma/client`.
   - Import the `db` instance from `~/server/db.ts`.
   - Import the `TeamMember` and `TeamMemberSchema` types from `~/types.ts`.

3. Define the API route handler function with the following signature:

   ```typescript
   export default async function handler(
     req: NextApiRequest,
     res: NextApiResponse
   ): Promise<void> {
     // Implementation goes here
   }
   ```

4. Inside the `handler` function, check if the request method is `GET`. If not, return a 405 (Method Not Allowed) status code with a JSON response containing an error message:

   ```typescript
   if (req.method !== "GET") {
     res.status(405).json({ error: "Method Not Allowed" });
     return;
   }
   ```

5. If the request method is `GET`, use the `db` instance to fetch all team members from the database:

   ```typescript
   const teamMembers = await db.teamMember.findMany();
   ```

6. Validate the `teamMembers` data using the `TeamMemberSchema` from `~/types.ts`. Use the `array()` method from Zod to create an array schema for the team members:

   ```typescript
   const teamMembersSchema = TeamMemberSchema.array();
   const validatedTeamMembers = teamMembersSchema.parse(teamMembers);
   ```

7. If the validation is successful, return a 200 (OK) status code with the JSON response containing the validated team members:

   ```typescript
   res.status(200).json(validatedTeamMembers);
   ```

8. Handle any errors that may occur during the database query or validation process. Return a 500 (Internal Server Error) status code with a JSON response containing an error message:

   ```typescript
   } catch (error) {
     res.status(500).json({ error: "Internal Server Error" });
   }
   ```

### Summary

To summarize, the `index.ts` file should contain the following:

1. Import the required modules and types.
2. Define the API route handler function with the correct signature.
3. Check if the request method is `GET`, and return a 405 status code if not.
4. Fetch all team members from the database using the `db` instance.
5. Validate the fetched team members using the `TeamMemberSchema`.
6. Return a 200 status code with the JSON response containing the validated team members.
7. Handle any errors and return a 500 status code with an error message.

Once this API endpoint is implemented, it will allow the front-end to fetch all team members and display them in the application.
